sycl/include/sycl/handler.hpp

```diff
@@ -2150,6 +2191,50 @@ class __SYCL_EXPORT handler {
     parallel_for_impl<KernelName>(Range, Properties, std::move(KernelFunc));
   }
 
+  template <typename KernelName, typename KernelType, int Dims,
+            typename PropertiesT>
+  void parallel_for_large_range(range<Dims> NumWorkItems, PropertiesT Props,
+                                KernelType KernelFunc) {
+    auto Dev = detail::getDeviceFromHandler(*this);
+    size_t WGSize = Dev.get_info<info::device::max_work_group_size>();
+    if (!range_size_fits_in_size_t(NumWorkItems))
+      throw sycl::exception(make_error_code(errc::runtime),
+                            "The total number of work-items in "
+                            "a range must fit within size_t");
+
+    size_t NOldWorkItems = NumWorkItems.size();
+    auto NWorkGroups = (std::numeric_limits<uint32_t>::max)() / WGSize;
```

Comment:
`info::device::max_work_group_size` returns _total_ maximum number of work-items in a work-group, but does not specify a _shape_ of a work-group.

There is also `info::device::max_work_item_sizes<N>` query, to understand per-dimension limits. It could be the case that `max_work_group_size` is `64`, but `max_work_item_sizes` is `{8, 8}`.

Note: there is also kernel-specific query for maximum number of work-items in a work-group: `info::kernel_device_specific::work_group_size`. If kernel is complex, it may have lower limits and generally supported by a device. However, I don't think that it is necessary to take it into account here, because usually extra limitation come from features like group operations (barrier) or local memory usage. Those features can't be used with a simple `parallel_for(range)`